### DISCUSSION TITLE
Remix nested layout (with actions) and `namedAction`

### DISCUSSION THREAD

**Discussion by jorisre** [sergiodxa/remix-utils#154] [Q&A] [answered]
Hi 👋🏻 

**Context:**

Let's imagine the following routes

```
/products
  $id.tsx => with action name `duplicate`
  $id/
     index.tsx
     detail.tsx => with action name `edit`
```

**Issue**

Both routes ($id and detail) use `namedAction`, when I submit duplicate action, the namedAction from detail.tsx route is triggered and throw a reference error `Action "duplicate" not found` and the parent route $id.tsx is triggered successfully. 

What's the correct way to manage the error?

**Comment by sergiodxa** [comment, marked as answer]
`namedAction` will work per route, so if you have a named action in one route and another named action in another route the library can't know based on the name what route to call, you POST to a route and the action function runs, then use `namedAction` to branch out inside the action function what it's doing.

in your case, if you want to run `duplicate` you need to send the post to `/products/:id` and not `/products/:id/detail`.

**Reply by jorisre** [reply]
Thanks, that's what I had identified after posting this message too. I changed the action path and it works ! thanks for your explanation
